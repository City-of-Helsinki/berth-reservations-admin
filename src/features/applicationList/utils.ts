import {
  BERTH_APPLICATIONS,
  BERTH_APPLICATIONS_berthApplications_edges as BERTH_APPLICATIONS_EDGE,
  BERTH_APPLICATIONS_berthApplications_edges_node as BERTH_APPLICATIONS_NODE,
} from './__generated__/BERTH_APPLICATIONS';
import { ApplicationStatus, LeaseStatus } from '../../@types/__generated__/globalTypes';

interface HarborChoice {
  harbor: string;
  harborName: string;
  priority: number;
}

interface Lease {
  berthNum: string;
  harborId: string;
  harborName: string;
  id: string;
  pierIdentifier: string;
  status: LeaseStatus;
}

interface BerthSwitch {
  berthNum: string;
  harborId: string;
  harborName: string;
  pierIdentifier: string;
  reason: string | null;
}

export interface ApplicationData {
  accessibilityRequired: boolean;
  berthSwitch: BerthSwitch | null;
  boatDraught: number | null;
  boatLength: number;
  boatModel: string;
  boatName: string;
  boatRegistrationNumber: string;
  boatType?: string | null;
  boatWeight: number | null;
  boatWidth: number;
  choices: Array<HarborChoice>;
  createdAt: string;
  customerId?: string;
  firstName: string;
  id: string;
  isSwitch: boolean;
  lastName: string;
  lease: Lease | null;
  municipality: string;
  queue: number;
  status: ApplicationStatus;
}

export const getBerthApplicationData = (data: BERTH_APPLICATIONS | undefined): ApplicationData[] => {
  const boatTypes = data?.boatTypes;

  return (
    data?.berthApplications?.edges.reduce<ApplicationData[]>((acc, edge) => {
      const {
        accessibilityRequired,
        berthSwitch,
        boatDraught,
        boatLength,
        boatModel,
        boatName,
        boatRegistrationNumber,
        boatType,
        boatWeight,
        boatWidth,
        createdAt,
        customer,
        firstName,
        harborChoices,
        id,
        lastName,
        lease,
        municipality,
        status,
      } = (edge as BERTH_APPLICATIONS_EDGE).node as BERTH_APPLICATIONS_NODE;

      const choices =
        harborChoices?.map((choice) => {
          return {
            priority: choice?.priority ?? Number.MAX_VALUE,
            harbor: choice?.harbor ?? '',
            harborName: choice?.harborName ?? '',
          };
        }) ?? [];

      let leaseProps: Lease | null = null;
      if (lease?.berth?.pier.properties?.harbor) {
        leaseProps = {
          berthNum: lease.berth.number || '',
          harborId: lease.berth.pier.properties.harbor.id,
          harborName: lease.berth.pier.properties.harbor.properties?.name || '',
          id: lease.id,
          pierIdentifier: lease.berth.pier.properties?.identifier || '',
          status: lease.status,
        };
      }

      const berthSwitchProps = berthSwitch && {
        berthNum: berthSwitch.berthNumber,
        harborId: berthSwitch.harbor,
        harborName: berthSwitch.harborName,
        pierIdentifier: berthSwitch.pier,
        reason: berthSwitch.reason?.title || null,
      };

      const applicationData = {
        accessibilityRequired,
        berthSwitch: berthSwitchProps,
        boatDraught,
        boatLength,
        boatModel,
        boatName,
        boatRegistrationNumber,
        boatType: boatTypes?.find(({ id }) => id === boatType)?.name,
        boatWeight,
        boatWidth,
        choices,
        createdAt,
        customerId: customer?.id,
        firstName,
        id,
        isSwitch: !!berthSwitch,
        lastName,
        lease: leaseProps,
        municipality,
        queue: 0, // TODO
        status,
      };

      return [...acc, applicationData];
    }, []) ?? []
  );
};
