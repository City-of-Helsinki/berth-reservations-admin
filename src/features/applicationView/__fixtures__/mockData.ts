import {
  INDIVIDUAL_APPLICATION,
  INDIVIDUAL_APPLICATION_additionalProducts as ADDITIONAL_PRODUCTS,
  INDIVIDUAL_APPLICATION_berthApplication as BERTH_APPLICATION,
  INDIVIDUAL_APPLICATION_berthApplication_berthSwitch as BERTH_SWITCH,
  INDIVIDUAL_APPLICATION_berthApplication_customer as CUSTOMER,
  INDIVIDUAL_APPLICATION_berthApplication_lease as LEASE,
  INDIVIDUAL_APPLICATION_boatTypes as BOAT_TYPES,
} from '../__generated__/INDIVIDUAL_APPLICATION';
import {
  AdditionalProductType,
  ApplicationStatus,
  BerthApplicationLanguage,
  BerthMooringType,
  CustomerGroup,
  InvoicingType,
  Language,
  PeriodType,
  PriceUnits,
  ProductServiceType,
} from '../../../@types/__generated__/globalTypes';

export const mockCustomer: CUSTOMER = {
  __typename: 'ProfileNode',
  comment: 'Comment',
  firstName: 'Testi',
  id: 'MOCK-CUSTOMER',
  invoicingType: InvoicingType.DIGITAL_INVOICE,
  language: Language.FINNISH,
  lastName: 'Testinen',
  organization: null,
  primaryAddress: { __typename: 'AddressNode', address: 'Testikatu 1', city: 'Helsinki', postalCode: '00100' },
  primaryEmail: { __typename: 'EmailNode', email: 'test@example.com' },
  primaryPhone: { __typename: 'PhoneNode', phone: '+358 00 000 0000' },
  customerGroup: CustomerGroup.PRIVATE,
};

export const mockBerthSwitch: BERTH_SWITCH = {
  __typename: 'BerthSwitchType',
  berthNumber: '1',
  harbor: 'MOCK-HARBOR',
  harborName: 'Test Harbor',
  id: 'MOCK-BERTH-SWITCH',
  pier: 'A',
  reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
};

export const mockLease: LEASE = {
  __typename: 'BerthLeaseNode',
  berth: {
    __typename: 'BerthNode',
    comment: '',
    depth: null,
    isAccessible: null,
    length: 5,
    mooringType: BerthMooringType.DINGHY_PLACE,
    number: 1,
    pier: {
      __typename: 'PierNode',
      properties: {
        __typename: 'PierProperties',
        electricity: false,
        gate: false,
        harbor: {
          __typename: 'HarborNode',
          id: 'MOCK-HARBOR',
          properties: { __typename: 'HarborProperties', name: 'Test Harbor' },
        },
        identifier: 'A',
        lighting: false,
        mooring: false,
        wasteCollection: false,
        water: false,
      },
    },
    width: 0,
  },
  id: 'MOCK-LEASE',
  order: null,
};

const additionalProducts: ADDITIONAL_PRODUCTS = {
  __typename: 'AdditionalProductNodeConnection',
  edges: [
    {
      __typename: 'AdditionalProductNodeEdge',
      node: {
        __typename: 'AdditionalProductNode',
        id: 'MOCK-ADDITIONAL-PRODUCT-NODE',
        period: PeriodType.MONTH,
        priceUnit: PriceUnits.AMOUNT,
        priceValue: '25.00',
        productType: AdditionalProductType.OPTIONAL_SERVICE,
        service: ProductServiceType.DINGHY_PLACE,
      },
    },
  ],
};

const berthApplication: BERTH_APPLICATION = {
  __typename: 'BerthApplicationNode',
  accessibilityRequired: false,
  address: 'Testikatu 1',
  berthSwitch: null,
  boatDraught: 1,
  boatLength: 1,
  boatModel: 'Marine',
  boatName: 'Cama la Yano',
  boatRegistrationNumber: '1234',
  boatType: '7',
  boatWeight: 120,
  boatWidth: 1,
  businessId: '',
  companyName: '',
  createdAt: '2020-07-10T08:04:36.062819+00:00',
  customer: null,
  email: 'test@example.com',
  firstName: 'Testi',
  harborChoices: [
    { harbor: 'MOCK-HARBOR-B', priority: 1, harborName: 'Saukonpaaden venesatama', __typename: 'HarborChoiceType' },
  ],
  id: 'MOCK-APPLICATION',
  language: BerthApplicationLanguage.FI,
  lastName: 'Testinen',
  lease: null,
  municipality: 'Helsinki',
  phoneNumber: '+358 00 000 0000',
  status: ApplicationStatus.PENDING,
  zipCode: '00100',
  acceptBoatingNewsletter: true,
  acceptFitnessNews: false,
  acceptLibraryNews: false,
  acceptOtherCultureNews: true,
  applicationCode: '1234',
};

const boatTypes: BOAT_TYPES[] = [
  { id: '1', name: 'Jollavene', __typename: 'BoatTypeType' },
  { id: '2', name: 'Soutuvene', __typename: 'BoatTypeType' },
  { id: '3', name: 'Perämoottorivene', __typename: 'BoatTypeType' },
  { id: '4', name: 'Sisäperämoottorivene', __typename: 'BoatTypeType' },
  { id: '5', name: 'Keskimoottorivene', __typename: 'BoatTypeType' },
  { id: '6', name: 'Purjevene / moottoripursi', __typename: 'BoatTypeType' },
  { id: '7', name: 'Troolari', __typename: 'BoatTypeType' },
  { id: '8', name: 'Suuri alus (yli 20t)', __typename: 'BoatTypeType' },
];

export const mockApplication: INDIVIDUAL_APPLICATION = {
  __typename: 'Query',
  additionalProducts: additionalProducts,
  berthApplication: berthApplication,
  boatTypes: boatTypes,
};
