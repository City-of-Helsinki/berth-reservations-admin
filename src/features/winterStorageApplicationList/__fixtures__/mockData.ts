import { WINTER_STORAGE_APPLICATIONS } from '../__generated__/WINTER_STORAGE_APPLICATIONS';
import { ApplicationStatus, WinterStorageApplicationAreaType } from '../../../@types/__generated__/globalTypes';

export const winterStorageApplicationMockData: WINTER_STORAGE_APPLICATIONS = {
  winterStorageApplications: {
    count: 83,
    edges: [
      {
        node: {
          id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTox',
          status: ApplicationStatus.PENDING,
          createdAt: '2019-05-06T10:54:21.746387+00:00',
          municipality: 'helsinki',
          customer: null,
          firstName: 'Testi',
          lastName: 'Testinen',
          boatType: '3',
          boatRegistrationNumber: 'test',
          boatWidth: 1,
          boatLength: 1,
          boatName: 'test',
          boatModel: 'test',
          areaType: WinterStorageApplicationAreaType.MARKED,
          winterStorageAreaChoices: [
            {
              priority: 1,
              winterStorageAreaName: 'Iso-Sarvasto',
              winterStorageArea: null,
              __typename: 'WinterStorageAreaChoiceType',
            },
            {
              priority: 2,
              winterStorageAreaName: 'Rajasaari',
              winterStorageArea: null,
              __typename: 'WinterStorageAreaChoiceType',
            },
          ],
          __typename: 'WinterStorageApplicationNode',
        },
        __typename: 'WinterStorageApplicationNodeEdge',
      },
      {
        node: {
          id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZToy',
          status: ApplicationStatus.PENDING,
          createdAt: '2019-05-06T12:30:10.157726+00:00',
          municipality: 'helsinki',
          customer: null,
          firstName: 'Testi',
          lastName: 'Testinen',
          boatType: '4',
          boatRegistrationNumber: 'test',
          boatWidth: 1,
          boatLength: 1,
          boatName: 'test',
          boatModel: 'test',
          areaType: WinterStorageApplicationAreaType.MARKED,
          winterStorageAreaChoices: [
            {
              priority: 1,
              winterStorageAreaName: 'Iso-Sarvasto',
              winterStorageArea: null,
              __typename: 'WinterStorageAreaChoiceType',
            },
          ],
          __typename: 'WinterStorageApplicationNode',
        },
        __typename: 'WinterStorageApplicationNodeEdge',
      },
      {
        node: {
          id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZToz',
          status: ApplicationStatus.PENDING,
          createdAt: '2019-05-06T12:30:27.609740+00:00',
          municipality: 'helsinki',
          customer: null,
          firstName: 'Testi',
          lastName: 'Testinen',
          boatType: '4',
          boatRegistrationNumber: 'test',
          boatWidth: 1,
          boatLength: 1,
          boatName: 'test',
          boatModel: 'test',
          areaType: WinterStorageApplicationAreaType.MARKED,
          winterStorageAreaChoices: [
            {
              priority: 1,
              winterStorageAreaName: 'Iso-Sarvasto',
              winterStorageArea: null,
              __typename: 'WinterStorageAreaChoiceType',
            },
          ],
          __typename: 'WinterStorageApplicationNode',
        },
        __typename: 'WinterStorageApplicationNodeEdge',
      },
      {
        node: {
          id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTo1',
          status: ApplicationStatus.PENDING,
          createdAt: '2019-05-06T13:07:28.521970+00:00',
          municipality: 'helsinki',
          customer: null,
          firstName: 'Testi',
          lastName: 'Testinen',
          boatType: '1',
          boatRegistrationNumber: 'test',
          boatWidth: 1,
          boatLength: 1,
          boatName: 'test',
          boatModel: 'test',
          areaType: WinterStorageApplicationAreaType.MARKED,
          winterStorageAreaChoices: null,
          __typename: 'WinterStorageApplicationNode',
        },
        __typename: 'WinterStorageApplicationNodeEdge',
      },
      {
        node: {
          id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTo2',
          status: ApplicationStatus.PENDING,
          createdAt: '2019-05-06T13:08:28.383257+00:00',
          municipality: 'helsinki',
          customer: null,
          firstName: 'Testi',
          lastName: 'Testinen',
          boatType: '1',
          boatRegistrationNumber: 'test',
          boatWidth: 1,
          boatLength: 1,
          boatName: 'test',
          boatModel: 'test',
          areaType: WinterStorageApplicationAreaType.MARKED,
          winterStorageAreaChoices: null,
          __typename: 'WinterStorageApplicationNode',
        },
        __typename: 'WinterStorageApplicationNodeEdge',
      },
      {
        node: {
          id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTo3',
          status: ApplicationStatus.PENDING,
          createdAt: '2019-05-06T13:08:38.133569+00:00',
          municipality: 'helsinki',
          customer: null,
          firstName: 'Testi',
          lastName: 'Testinen',
          boatType: '1',
          boatRegistrationNumber: 'test',
          boatWidth: 1,
          boatLength: 1,
          boatName: 'test',
          boatModel: 'test',
          areaType: WinterStorageApplicationAreaType.MARKED,
          winterStorageAreaChoices: null,
          __typename: 'WinterStorageApplicationNode',
        },
        __typename: 'WinterStorageApplicationNodeEdge',
      },
      {
        node: {
          id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTo4',
          status: ApplicationStatus.PENDING,
          createdAt: '2019-05-06T13:09:44.953722+00:00',
          municipality: 'helsinki',
          customer: null,
          firstName: 'Testi',
          lastName: 'Testinen',
          boatType: '1',
          boatRegistrationNumber: 'test',
          boatWidth: 1,
          boatLength: 1,
          boatName: 'test',
          boatModel: 'test',
          areaType: WinterStorageApplicationAreaType.MARKED,
          winterStorageAreaChoices: null,
          __typename: 'WinterStorageApplicationNode',
        },
        __typename: 'WinterStorageApplicationNodeEdge',
      },
      {
        node: {
          id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTo5',
          status: ApplicationStatus.PENDING,
          createdAt: '2019-05-06T13:09:56.820368+00:00',
          municipality: 'helsinki',
          customer: null,
          firstName: 'Testi',
          lastName: 'Testinen',
          boatType: '1',
          boatRegistrationNumber: 'test',
          boatWidth: 1,
          boatLength: 1,
          boatName: 'test',
          boatModel: 'test',
          areaType: WinterStorageApplicationAreaType.MARKED,
          winterStorageAreaChoices: null,
          __typename: 'WinterStorageApplicationNode',
        },
        __typename: 'WinterStorageApplicationNodeEdge',
      },
      {
        node: {
          id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZToxMA==',
          status: ApplicationStatus.PENDING,
          createdAt: '2019-05-06T13:12:02.766890+00:00',
          municipality: 'helsinki',
          customer: null,
          firstName: 'Testi',
          lastName: 'Testinen',
          boatType: '1',
          boatRegistrationNumber: 'test',
          boatWidth: 1,
          boatLength: 1,
          boatName: 'test',
          boatModel: 'test',
          areaType: WinterStorageApplicationAreaType.MARKED,
          winterStorageAreaChoices: null,
          __typename: 'WinterStorageApplicationNode',
        },
        __typename: 'WinterStorageApplicationNodeEdge',
      },
      {
        node: {
          id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZToxMQ==',
          status: ApplicationStatus.PENDING,
          createdAt: '2019-05-07T18:29:18.066260+00:00',
          municipality: 'helsinki',
          customer: null,
          firstName: 'Testi',
          lastName: 'Testinen',
          boatType: '1',
          boatRegistrationNumber: 'test',
          boatWidth: 1,
          boatLength: 1,
          boatName: 'test',
          boatModel: 'test',
          areaType: WinterStorageApplicationAreaType.MARKED,
          winterStorageAreaChoices: null,
          __typename: 'WinterStorageApplicationNode',
        },
        __typename: 'WinterStorageApplicationNodeEdge',
      },
    ],
    __typename: 'WinterStorageApplicationNodeConnection',
  },
  boatTypes: [
    {
      id: '1',
      name: 'Jollavene',
      __typename: 'BoatTypeType',
    },
    {
      id: '2',
      name: 'Soutuvene',
      __typename: 'BoatTypeType',
    },
    {
      id: '3',
      name: 'Perämoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '4',
      name: 'Sisäperämoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '5',
      name: 'Keskimoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '6',
      name: 'Purjevene / moottoripursi',
      __typename: 'BoatTypeType',
    },
    {
      id: '7',
      name: 'Troolari',
      __typename: 'BoatTypeType',
    },
    {
      id: '8',
      name: 'Suuri alus (yli 20t)',
      __typename: 'BoatTypeType',
    },
  ],
};
