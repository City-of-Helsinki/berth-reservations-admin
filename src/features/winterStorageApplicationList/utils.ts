import { ApplicationStatus } from '../../@types/__generated__/globalTypes';
import { WINTER_STORAGE_APPLICATIONS } from './__generated__/WINTER_STORAGE_APPLICATIONS';
import { ApplicationTypeEnum } from '../../common/applicationDetails/types';

interface WinterStorageAreaChoice {
  priority: number;
  winterStorageArea: string;
  winterStorageAreaName: string;
}

export type WinterStorageApplication = {
  applicationCode: string;
  applicationType: ApplicationTypeEnum;
  boatLength: number;
  boatModel: string;
  boatName: string;
  boatRegistrationNumber: string;
  boatType?: string | null;
  boatWidth: number;
  choices: WinterStorageAreaChoice[];
  createdAt: string;
  customerId?: string;
  firstName: string;
  id: string;
  lastName: string;
  municipality: string;
  queue: number;
  status: ApplicationStatus;
};

export const getWinterStorageApplicationData = (
  data: WINTER_STORAGE_APPLICATIONS | undefined
): WinterStorageApplication[] => {
  const boatTypes = data?.boatTypes;

  return (
    data?.winterStorageApplications?.edges.reduce<WinterStorageApplication[]>((acc, edge) => {
      if (!edge?.node) return acc;

      const {
        applicationCode,
        boatLength,
        boatModel,
        boatName,
        boatRegistrationNumber,
        boatType,
        boatWidth,
        createdAt,
        customer,
        firstName,
        id,
        lastName,
        municipality,
        status,
        winterStorageAreaChoices,
      } = edge.node;

      const choices =
        winterStorageAreaChoices?.map((choice) => {
          return {
            priority: choice?.priority ?? Number.MAX_VALUE,
            winterStorageAreaName: choice?.winterStorageArea.properties?.name ?? '',
            winterStorageArea: choice?.winterStorageArea.id ?? '',
          };
        }) ?? [];

      // TODO: lease

      const applicationData = {
        applicationCode,
        applicationType: ApplicationTypeEnum.WINTER_STORAGE,
        boatLength,
        boatModel,
        boatName,
        boatRegistrationNumber,
        boatType: boatTypes?.find(({ id }) => id === boatType)?.name,
        boatWidth,
        choices,
        createdAt,
        customerId: customer?.id,
        firstName,
        id,
        lastName,
        municipality,
        queue: 0, // TODO,
        status,
      };

      return [...acc, applicationData];
    }, []) ?? []
  );
};
