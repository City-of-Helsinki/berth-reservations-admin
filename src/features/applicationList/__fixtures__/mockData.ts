import { BERTH_APPLICATIONS } from '../__generated__/BERTH_APPLICATIONS';
import { getBerthApplicationData } from '../utils';
import { ApplicationStatus, LeaseStatus } from '../../../@types/__generated__/globalTypes';

export const mockData: BERTH_APPLICATIONS = {
  berthApplications: {
    __typename: 'BerthApplicationNodeConnection',
    count: 3,
    edges: [
      // Full structure
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '1',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-0',
            pier: 'A',
            reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano',
          boatDraught: 5,
          boatRegistrationNumber: 'A 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Matti',
          harborChoices: [{ __typename: 'HarborChoiceType', harbor: '1', harborName: '1', priority: 0 }],
          id: 'MOCK-APPLICATION-0',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-0',
                    properties: { __typename: 'HarborProperties', name: 'Testisatama' },
                  },
                  identifier: 'G',
                },
              },
            },
            id: 'MOCK-LEASE-0',
            status: LeaseStatus.OFFERED,
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Most structure and nulled fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '2',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-1',
            pier: 'A',
            reason: null,
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano II',
          boatDraught: 5,
          boatRegistrationNumber: 'B 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Maija',
          harborChoices: null,
          id: 'MOCK-APPLICATION-1',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            id: 'MOCK-LEASE-1',
            status: LeaseStatus.OFFERED,
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  identifier: 'Pier Brosnan',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-1',
                    properties: {
                      __typename: 'HarborProperties',
                      name: null,
                    },
                  },
                },
              },
            },
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Minimum fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: null,
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano 3-D',
          boatDraught: 5,
          boatRegistrationNumber: 'C 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE-0' },
          firstName: 'Matti',
          harborChoices: [
            { __typename: 'HarborChoiceType', harbor: 'MOCK-HARBOR-0', harborName: 'Testisatama', priority: 0 },
          ],
          id: 'MOCK-APPLICATION-2',
          lastName: 'Möttönen',
          lease: null,
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
    ],
  },
  boatTypes: [{ __typename: 'BoatTypeType', id: '1', name: 'Purjevene' }],
};

export const mockTableData = getBerthApplicationData(mockData);
