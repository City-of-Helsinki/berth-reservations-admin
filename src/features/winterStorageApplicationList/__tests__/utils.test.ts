import { getWinterStorageApplicationData } from '../utils';
import { winterStorageApplicationMockData } from '../__fixtures__/mockData';

describe('winterStorageApplicationList utils', () => {
  describe('getWinterStorageApplicationData', () => {
    it('correctly parses response data', () => {
      const applications = getWinterStorageApplicationData(winterStorageApplicationMockData);
      expect(applications).toMatchSnapshot();
    });
  });
});
