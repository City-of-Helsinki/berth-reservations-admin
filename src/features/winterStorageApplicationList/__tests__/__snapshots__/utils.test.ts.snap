// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`winterStorageApplicationList utils getWinterStorageApplicationData correctly parses response data 1`] = `
Array [
  Object {
    "applicationCode": "",
    "boatLength": 1,
    "boatModel": "test",
    "boatName": "test",
    "boatRegistrationNumber": "test",
    "boatType": "Perämoottorivene",
    "boatWidth": 1,
    "choices": Array [
      Object {
        "priority": 1,
        "winterStorageArea": "Iso-Sarvasto",
        "winterStorageAreaName": "Iso-Sarvasto",
      },
      Object {
        "priority": 2,
        "winterStorageArea": "Rajasaari",
        "winterStorageAreaName": "Rajasaari",
      },
    ],
    "createdAt": "2019-05-06T10:54:21.746387+00:00",
    "customerId": undefined,
    "firstName": "Testi",
    "id": "V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTox",
    "lastName": "Testinen",
    "municipality": "helsinki",
    "queue": 0,
    "status": "PENDING",
  },
  Object {
    "applicationCode": "code",
    "boatLength": 1,
    "boatModel": "test",
    "boatName": "test",
    "boatRegistrationNumber": "test",
    "boatType": "Sisäperämoottorivene",
    "boatWidth": 1,
    "choices": Array [
      Object {
        "priority": 1,
        "winterStorageArea": "Iso-Sarvasto",
        "winterStorageAreaName": "Iso-Sarvasto",
      },
    ],
    "createdAt": "2019-05-06T12:30:10.157726+00:00",
    "customerId": undefined,
    "firstName": "Testi",
    "id": "V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZToy",
    "lastName": "Testinen",
    "municipality": "helsinki",
    "queue": 0,
    "status": "PENDING",
  },
  Object {
    "applicationCode": "",
    "boatLength": 1,
    "boatModel": "test",
    "boatName": "test",
    "boatRegistrationNumber": "test",
    "boatType": "Sisäperämoottorivene",
    "boatWidth": 1,
    "choices": Array [
      Object {
        "priority": 1,
        "winterStorageArea": "Iso-Sarvasto",
        "winterStorageAreaName": "Iso-Sarvasto",
      },
    ],
    "createdAt": "2019-05-06T12:30:27.609740+00:00",
    "customerId": undefined,
    "firstName": "Testi",
    "id": "V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZToz",
    "lastName": "Testinen",
    "municipality": "helsinki",
    "queue": 0,
    "status": "PENDING",
  },
  Object {
    "applicationCode": "",
    "boatLength": 1,
    "boatModel": "test",
    "boatName": "test",
    "boatRegistrationNumber": "test",
    "boatType": "Jollavene",
    "boatWidth": 1,
    "choices": Array [],
    "createdAt": "2019-05-06T13:07:28.521970+00:00",
    "customerId": undefined,
    "firstName": "Testi",
    "id": "V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTo1",
    "lastName": "Testinen",
    "municipality": "helsinki",
    "queue": 0,
    "status": "PENDING",
  },
  Object {
    "applicationCode": "",
    "boatLength": 1,
    "boatModel": "test",
    "boatName": "test",
    "boatRegistrationNumber": "test",
    "boatType": "Jollavene",
    "boatWidth": 1,
    "choices": Array [],
    "createdAt": "2019-05-06T13:08:28.383257+00:00",
    "customerId": undefined,
    "firstName": "Testi",
    "id": "V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTo2",
    "lastName": "Testinen",
    "municipality": "helsinki",
    "queue": 0,
    "status": "PENDING",
  },
  Object {
    "applicationCode": "",
    "boatLength": 1,
    "boatModel": "test",
    "boatName": "test",
    "boatRegistrationNumber": "test",
    "boatType": "Jollavene",
    "boatWidth": 1,
    "choices": Array [],
    "createdAt": "2019-05-06T13:08:38.133569+00:00",
    "customerId": undefined,
    "firstName": "Testi",
    "id": "V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTo3",
    "lastName": "Testinen",
    "municipality": "helsinki",
    "queue": 0,
    "status": "PENDING",
  },
  Object {
    "applicationCode": "code",
    "boatLength": 1,
    "boatModel": "test",
    "boatName": "test",
    "boatRegistrationNumber": "test",
    "boatType": "Jollavene",
    "boatWidth": 1,
    "choices": Array [],
    "createdAt": "2019-05-06T13:09:44.953722+00:00",
    "customerId": undefined,
    "firstName": "Testi",
    "id": "V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTo4",
    "lastName": "Testinen",
    "municipality": "helsinki",
    "queue": 0,
    "status": "PENDING",
  },
  Object {
    "applicationCode": "",
    "boatLength": 1,
    "boatModel": "test",
    "boatName": "test",
    "boatRegistrationNumber": "test",
    "boatType": "Jollavene",
    "boatWidth": 1,
    "choices": Array [],
    "createdAt": "2019-05-06T13:09:56.820368+00:00",
    "customerId": undefined,
    "firstName": "Testi",
    "id": "V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTo5",
    "lastName": "Testinen",
    "municipality": "helsinki",
    "queue": 0,
    "status": "PENDING",
  },
  Object {
    "applicationCode": "",
    "boatLength": 1,
    "boatModel": "test",
    "boatName": "test",
    "boatRegistrationNumber": "test",
    "boatType": "Jollavene",
    "boatWidth": 1,
    "choices": Array [],
    "createdAt": "2019-05-06T13:12:02.766890+00:00",
    "customerId": undefined,
    "firstName": "Testi",
    "id": "V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZToxMA==",
    "lastName": "Testinen",
    "municipality": "helsinki",
    "queue": 0,
    "status": "PENDING",
  },
  Object {
    "applicationCode": "",
    "boatLength": 1,
    "boatModel": "test",
    "boatName": "test",
    "boatRegistrationNumber": "test",
    "boatType": "Jollavene",
    "boatWidth": 1,
    "choices": Array [],
    "createdAt": "2019-05-07T18:29:18.066260+00:00",
    "customerId": undefined,
    "firstName": "Testi",
    "id": "V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZToxMQ==",
    "lastName": "Testinen",
    "municipality": "helsinki",
    "queue": 0,
    "status": "PENDING",
  },
]
`;
